src/translations.jl

```diff
@@ -50,11 +50,28 @@ with non-zero `thresh`, it is not permissible to "align" the images by shifting
 function qd_translate(fixed, moving, mxshift;
                       thresh=0.1*sum(abs2.(fixed[.!(isnan.(fixed))])),
                       initial_tfm=IdentityTransformation(),
-                      minwidth=fill(0.01, ndims(fixed)), print_interval=100, kwargs...)
+                      minwidth=fill(0.01, ndims(fixed)), print_interval=100, crop=false, kwargs...)
     fixed, moving = float(fixed), float(moving)
     print_interval < typemax(Int) && print("Running coarse step\n")
-    best_shft, mm = best_shift(fixed, moving, mxshift, thresh; normalization=:intensity, initial_tfm=initial_tfm)
+    if crop
+        #we enforce that moving is always bigger than fixed by amount 2*(maxshift+1) (the +1 is for the fine step)
+        sz = size(fixed) .- (2 .* mxshift)
+        if any(size(moving) .< (2 .* (mxshift.+1)))
+            error("Moving image size must be at least 2 * (mxshift+1) when crop_edges is set to true")
+        end
+        cropped_inds_f = crop_rng.(UnitRange.(axes(fixed)), mxshift)
+        cropped_inds_m = crop_rng.(UnitRange.(axes(moving)), mxshift)
+        moving_inner = view(moving, cropped_inds_m...)
+        fixed_inner = view(fixed, cropped_inds_f...)
+        moving_fine = OffsetArray(moving, (-1 .* mxshift)...)
+    else
+        fixed_inner = fixed
+        moving_inner = moving_fine = moving
+    end
+    best_shft, mm = best_shift(fixed_inner, moving_inner, mxshift, thresh; normalization=:intensity, initial_tfm=initial_tfm)
     tfm_coarse = initial_tfm ∘ Translation(best_shft)
     print_interval < typemax(Int) && print("Running fine step\n")
-    return qd_translate_fine(fixed, moving; initial_tfm=tfm_coarse, thresh=thresh, minwidth=minwidth, print_interval=print_interval, kwargs...)
+    return qd_translate_fine(fixed_inner, moving_fine; initial_tfm=tfm_coarse, thresh=thresh, minwidth=minwidth, print_interval=print_interval, kwargs...)
 end
+
+crop_rng(rng::UnitRange{Int}, amt::Int) = (first(rng)+amt):(last(rng)-amt)
```

Comment:
Why does this need to be restricted to `UnitRange`? How about `::AbstractUnitRange`? Then you can also get rid of the `UnitRange.(axes(...` above.

Reply:
Ah yes that would be more elegant.  I guess I wasn't sure whether this would hold for any AbstractUnitRange.  As long as `first` and `last` are defined I suppose it should.  And if they're not defined this will throw a useful error, so it sounds like a good idea to me.  I'll update.